### PR TITLE
Use `addonAfter` instead of `suffix` when use `enterButton` in Input.Search

### PR DESCRIPTION
**Pull request by zombieJ**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

#14450
  
### API Realization (Optional if not new feature)

Not change

### What's the effect? (Optional if not new feature)

* May affect some customize style on addonAfter and suffix
* Move `className` from inner input to top wrapper when with addon

### Changelog description (Optional if not new feature)

* Fix Input.Search style issue. Move `className` from inner input to top wrapper when with addon.
* 修复 Input.Search 样式问题。在 addon 下，将 `className` 从 input 改到最顶层元素上。

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input/Input.tsx [hunk 1]
```diff
@@ -185,38 +189,37 @@ class Input extends React.Component<InputProps, any> {
...
+    const mergedGroupClassName = classNames(_internalGroupClassName, `${prefixCls}-group-wrapper`, {
+      [`${prefixCls}-group-wrapper-sm`]: size === 'small',
+      [`${prefixCls}-group-wrapper-lg`]: size === 'large',
     });
 
     // Need another wrapper for changing display:table to display:inline-block
     // and put style prop in wrapper
     return (
-      <span className={groupClassName} style={props.style}>
+      <span className={mergedGroupClassName} style={style}>
```
[the remark this answers is not in the document]
Comment (zombieJ) [reply components/input/Input.tsx]: 这个 `className` 不是 `props.className`，`props.className` 现在是放在 input 上的。

[the remark this answers is not in the document]
Comment (afc163) [reply components/input/Input.tsx]: > props.className 现在是放在 input 上的。

这应该是有问题的。

也导致我们不得不加个奇怪 `_internalGroupClassName`，我建议一起修正了。

[the remark this answers is not in the document]
Comment (zombieJ) [reply components/input/Input.tsx]: 是个 break change，用 className 修改的 border 之类的属性会失效掉。不过好在还可以通过 selector 对下去。改完 changelog 里需要标注一下。

**Comment by zombieJ** [comment]
Updated. Pls check.

File: components/input/Input.tsx [hunk 2]
```diff
@@ -265,6 +263,7 @@ class Input extends React.Component<InputProps, any> {
       // Input elements must be either controlled or uncontrolled,
       // specify either the value prop, or the defaultValue prop, but not both.
       'defaultValue',
+      '_internalGroupClassName',
```
Comment (afc163) [inline components/input/Input.tsx]: 这个不需要了。

Comment (zombieJ) [reply components/input/Input.tsx]: 好了，看看还有没有其他需要改进的~

**Review by afc163 — APPROVED** [review]

**Pull request merged**